### DISCUSSION TITLE
Count not updating

### DISCUSSION THREAD

**Discussion by Ryurcia** [supabase/supabase#20914] [Questions]
I have a function that retrieves follower count:
```
const getFollowerStat = async (userId: string) => {
  const supabase = createServerComponentClient<Database>({cookies});
  const {
    count,
    error
  } = await supabase.from('Follow').select('*', {count: 'exact',head:true}).eq('following_id', userId);

  if (error) throw error.message;
  if (!count) console.log('No Data');

  console.log(count);
  return count;
};
```
But whenever I execute it for a user, it always returns 0. But when I tried making the same query in the SQL Editor on Supabase, it returns the right count.  Any help is appreciated!

**Comment by GaryAustin1** [comment]
RLS would be the first thing to look at.
Then if next.js caching they do on URLs would be then next...

**Reply by Ryurcia** [reply]
I just disabled RLS for the table so its working fine now. But, my policy for when it was enabled was to allow all actions for authenticated users, it did not work even though the user is authenticated.

**Reply by GaryAustin1** [reply]
Then you likely do not really have a user session in the client.  You could set your policies to ANON role and true.  If that works then you don't have a user session right in that server component client.
